spec/src/main/asciidoc/jsonb.adoc

```diff
@@ -649,7 +649,41 @@ Mapping between parameters of constructor/factory method annotated with `JsonbCr
 
 In case `JsonbProperty` annotation on parameters is not used, parameters should be mapped from JSON fields with the same name. In this case the proper mapping is NOT guaranteed.
 
-In case a field required for a parameter mapping doesn’t exist in JSON document, `JsonbException` MUST be thrown.
+In case a field required for a parameter mapping doesn't exist in JSON document, `JsonbException` MUST be thrown unless it is marked as an optional. See <<optional-creator-params, Optional JsonbCreator parameters>>. All the parameters are treated as optional by default.
+
+[[optional-creator-params]]
+==== Optional JsonbCreator parameters
+Optional creator parameter is parameter which is not required to be present in the JSON document and if it is missing, no exception will be thrown.
+
+Implementation must allow setting all parameters to be required of a certain creator. This is achieved by adding `JsonbRequired` annotation to the creator method/constructor. If value of this annotation is set to false, all parameters of the creator will be treated as optional.
+
+Implementation must allow making only certain parameters as optional/required. Parameter can be declared as required if annotated with `JsonbRequired`.
+
+Parameter can be declared as optional by using:
+
+* Parameter annotated with `JsonbRequired` and having value set to `false`
+* Parameter is `Optional` type
+
+If property has explicit value `null` set in the JSON document and parameter is primitive type, implementation must throw an exception.
+
+==== Optional parameter values
+Whenever a property is missing from JSON document and is marked as optional, proper default value should be used. If the parameter is type of any Object except `Optional` and its variations, `null` value is used. In case of `Optional` and its variations, corresponding empty object must be used.
+
+Primitive types cannot accept null values, and therefore it is required to provide corresponding value, as listed in the following table.
```

Comment:
Primitive types cannot accept null values, so a corresponding value is required as listed in the following table: